.Documentation/articles/developer/07-run-a-course.md

```diff
@@ -1,3 +1,93 @@
+# Create a Course
+
+## Entity Factory
+
+The `EntityFactory` is a helper class that allows you to dynamically create basic elements from a course. It is part of the Innoactive Creator's API.
+
+It supports the creation of steps (`IStep`), transitions (`ITransition`), chapters (`IChapter`), and courses (`ICourse`). A course can be edited as much as possible only if it is not being played.
```

Comment:
```suggestion
It supports the creation of steps (`IStep`), transitions (`ITransition`), chapters (`IChapter`), and courses (`ICourse`). You can modify the course as much as you want until the application starts executing it.
```

Reply:
Is it actually true? Can't you add new steps to the course as it plays out?

Reply:
Actually, I am not sure, I thought that even if it would be possible it might mess with the system, but let me try it out.

Reply:
Apparently it is possible to edit the course while it is been playing, at least no errors are thrown, but I found a lot of difficulties for adding more steps, for instance after adding only one step the course executes that new dynamic step but it gets stuck in the Activation phase, not sure if it has to do with the course being played on runtime or it actually will happen even if the course is edited or created dynamically 😕